src/test/java/com/rewedigital/composer/composing/ValidatingContentFetcherTest.java

```diff
@@ -72,16 +81,43 @@ public void forwardsSession() throws Exception {
         final Client client = mock(Client.class);
         when(client.send(any())).thenReturn(aResponse(""));
         final SessionRoot session = session("x-rd-key", "value");
-        new ValidatingContentFetcher(client, emptyMap(), session).fetch("/some/path", "", aStep()).get();
+        new ValidatingContentFetcher(client, emptyMap(), session)
+            .fetch(FetchContext.of("/some/path", "", defaultTimeout), aStep())
+            .get();
         verify(client).send(aRequestWithSession("x-rd-key", "value"));
     }
 
+    @Test
+    public void validatingContentFetcherAppliesTtl() throws Exception {
+        final StubClient client = aStubClient();
+        final Optional<Duration> timeout = Optional.of(Duration.ofMillis(200));
+
+        new ValidatingContentFetcher(client, emptyMap(), emptySession)
+            .fetch(FetchContext.of("path", "fallback", timeout), aStep())
+            .get();
+
+        assertThat(client.sentRequests()).isNotEmpty().allSatisfy(r -> ttlIsSet(r, timeout));
+    }
+
+    private void ttlIsSet(final Request request, final Optional<Duration> timeout) {
+        assertThat(request.ttl()).isNotEmpty();
```

Comment:
You can use `assertThat(request.ttl()).contains("...")` to assert that some optional is present with some content

Reply:
Done